cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/web/WhoAmIServlet.java

```diff
@@ -206,15 +201,12 @@ public AuthMethod getAuthMethod(Subject subject)
      * @param principal     The HttpPrincipal instance.
      * @param authMethod    The authMethod
      */
-    void redirect(HttpServletResponse response, Principal principal, AuthMethod authMethod) throws IOException
+    void redirect(StringBuffer requestURL, HttpServletResponse response, Principal principal, AuthMethod authMethod) throws IOException
     {
-        final RegistryClient registryClient = getRegistryClient();
-
-        URI umsServiceURI = getServiceURI(Standards.UMS_WHOAMI_01);
-        log.debug("ums service uri: " + umsServiceURI);
-
-        final URL serviceURL = registryClient.getServiceURL(umsServiceURI, Standards.UMS_USERS_01, authMethod);
-        final URL redirectURL = new URL(serviceURL.toExternalForm() + USER_GET_PATH);
+        // TODO: Look for a flexible way (vs currently hardcoded) to append '/users' to the baseServiceURL
+        String baseServiceURL = requestURL.substring(0, requestURL.lastIndexOf("/"));
```

Comment:
This might be crazy, but what about simply replacing WHOAMI with USERS since both are constants? That should work even if the service is renamed (see main comment).

Reply:
Replaced "WHOAMI" constant with "USERS".

Reply:
oops, what I meant was that finding the base by looking for the last `/` is potentially fragile; I was suggesting something like
```
String usersEndpoint = requestURL.toString().replace(EndpointConstants.WHOAMI, EndpointConstants.USERS);
```
since /whoami and /users are always siblings. However, you'd also have to be careful to strip off any cruft from the end (bogus query string or fragments, etc) so doing the replace will work in the correct usage but potentially not if someone is messing around.

instead of looking for the `lastIndexOf("/") you could look for the `firstIndexOf(EndpointConstants.WHOAMI)` as a more robust indicator of the end of the base service URL. Maybe `firstIndexOf("/" + EndpointConstants.WHOAMI)` depending to get the same index as before...

Reply:
My misunderstanding. Using first index now.